asv/plugins/mamba.py

```diff
@@ -155,6 +155,12 @@ def _setup(self):
                 except KeyError:
                     raise KeyError("Only pip is supported as a secondary key")
         mamba_pkgs += mamba_args
+        # Changed in v0.6.5, gh-1294
+        # previously, the user provided environment was assumed to handle the python version
+        mamba_pkgs = [
+            util.replace_python_version(pkg, self._python) for pkg in mamba_pkgs
+        ]
```

Comment:
@HaoZeke @mattip sorry if I'm missing something, but it looks like if you do not have an environment file, the python version specified in asv.conf.json no longer gets passed to mamba after this change. (ie if there is no python= constraint before this line runs, we never add one) Perhaps we should restore the `python={self._python}` in the initialization of mamba_pkgs?